src/Storages/ObjectStorage/DataLakes/Iceberg/Utils.cpp

```diff
@@ -545,11 +547,12 @@ std::pair<Poco::JSON::Object::Ptr, String> createEmptyMetadataFile(
     schema_representation->set(Iceberg::f_schema_id, 0);
 
     Poco::JSON::Array::Ptr schema_fields = new Poco::JSON::Array;
-    Int32 iter = 0;
+    Int32 iter = static_cast<Int32>(columns.size());
+    Int32 iter_for_initial_columns = 0;
     for (const auto & column : columns)
     {
         Poco::JSON::Object::Ptr field = new Poco::JSON::Object;
-        field->set(Iceberg::f_id, ++iter);
+        field->set(Iceberg::f_id, ++iter_for_initial_columns);
```

Comment:
Are you sure we now need to assign the same `iter` value for all `column_name_to_source_id[column.name]`?